pkg/reconciler/subscription/subscription.go

```diff
@@ -229,11 +229,13 @@ func (r *Reconciler) resolveSubscriber(ctx context.Context, subscription *v1.Sub
 		subscriberURI := subscriberAddr.URL
 		// If there is a change in resolved URI, log it.
 		if subscription.Status.PhysicalSubscription.SubscriberURI == nil || subscription.Status.PhysicalSubscription.SubscriberURI.String() != subscriberURI.String() {
-			logging.FromContext(ctx).Debugw("Resolved Subscriber", zap.String("subscriberURI", subscriberURI.String()))
+			logging.FromContext(ctx).Debugw("Resolved Subscriber", zap.String("subscriberURI", subscriberURI.String()), zap.Stringp("subscriberCACerts", subscriberAddr.CACerts))
 			subscription.Status.PhysicalSubscription.SubscriberURI = subscriberURI
+			subscription.Status.PhysicalSubscription.SubscriberCACerts = subscriberAddr.CACerts
 		}
```

Comment:
Is the `subscription.Status.PhysicalSubscription.SubscriberURI == nil || subscription.Status.PhysicalSubscription.SubscriberURI.String() != subscriberURI.String()` condition still appropriate?

Reply:
Can we add some unit tests?

Reply:
> Is the subscription.Status.PhysicalSubscription.SubscriberURI == nil || subscription.Status.PhysicalSubscription.SubscriberURI.String() != subscriberURI.String() condition still appropriate?

I removed the check completely now, as this should be the same regardless of anything changes IIUC (or is only something set in the `subscription.Status` if something changes (e.g. if this is converted to a patch)?).

> Can we add some unit tests?
:+1: I added some